modules/c++/dbi/source/DatabaseClientFactory.cpp

```diff
@@ -24,7 +24,9 @@
 #include "dbi/MySQLConnection.h"
 #include "dbi/PgSQLConnection.h"
 #include "dbi/OracleConnection.h"
-
+#include "config/compiler_extensions.h"
+CODA_OSS_disable_warning_push
+CODA_OSS_disable_warning(4702)
```

Comment:
To make this work properly, you'll need to add something like the following to `modules/c++/config/include/config/compiler_extensions.h`
```c++
// in the msvc section...
#define CODA_OSS_DISABLE_UNREACHABLE_CODE CODA_OSS_disable_warning(4702)
// in the gcc/clang section and the #else ...
#define CODA_OSS_DISABLE_UNREACHABLE_CODE
```
and then use `CODA_OSS_DISABLE_UNREACHABLE_CODE` in this file